packages/js/README.md

```diff
@@ -136,6 +136,79 @@ MatomoInstance.trackSiteSearch({
 })
 ```
 
+If you enabled Ecommerce for your site, you can track purchased orders like this:
```

Comment:
It's super nice that you wrote some documentation, however if we do this here we would have to add documentation for every Matamo feature in the README which would make things a bit cluttered.

Could you add this kind of documentation as JSDoc to the methods you introduced? That way we can generate API docs from code in the future.

Reply:
ok.

currently moving the parts from the readme into `MatomoTracker.js` and wondering if it is helpful to have the examples within source code. it seems cluttered and also makes assumptions of how these will be rendered (e.g. github flavoured markdown to have syntax highlighting). 

since this is typescript, most arguments to the added functions have a type signature, adding an explicit `@param` to those also seems redundant. shall I just add the short description to keep the comments in the style of the current (other) functions (e.g. `trackPageView`)?

Reply:
see the updated pr for minimal docs (i.e. no formatting in comments since other comments also don't)

Reply:
Looks good. Yeah, the current documentation can be improved and is something we will do in the future.

Reply:
I agree with you, we already have type definitions that act like documentation on itself. We currently do not use an API docs generator so I think it's good now, right @jonkoops ?

Reply:
@chrisvanmook Yeah, I am talking about adding JSDoc annotations to our methods. Which will show up in any modern IDE such as VSCode and WebStorm. But this is out of scope for this PR.